File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxDepth(TreeNode root) {
        // post order
        return height(root);
        
    }
    public int height(TreeNode node){
        // base case
        if(node == null) return 0;
        
        // left 
        int left = height(node.left);
        
        // right
        int right = height(node.right);
        
        // node
        return Math.max(right, left) + 1;
    }
    
}

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        return self.height(root)
    
    def height(self, node: Optional[TreeNode]) -> int:
        #base case
        if node is None:
            return 0
        left = self.height(node.left)
        right = self.height(node.right)
        return max(left, right) + 1
        